internal/tarwriter/tarwriter.go

```diff
@@ -77,7 +77,16 @@ func (tw *tarWriter) Flush() error {
 	}
 	info, _ := os.Stat(fname)
 
-	b, uploadErr := upload.Upload(tw.uploadURL, fname, "application/vnd.redhat.topological-inventory.filename+tgz", tw.metadata)
+	if sha == tw.input.PreviousSHA && info.Size() == tw.input.PreviousSize {
+		err = tw.task.Update(map[string]interface{}{"state": "completed", "status": "unchanged", "message": "Nothing changes since last refresh"})
+		if err != nil {
+			tw.glog.Errorf("Error updating task: %v", err)
+			return err
+		}
+		return nil
+	}
+
+	b, uploadErr := upload.Upload(tw.input.UploadURL, fname, "application/vnd.redhat.topological-inventory.filename+tgz", tw.metadata)
```

Comment:
@bzwei Should this be catalog now. We had changed it in this PR https://github.com/RedHatInsights/rhc-worker-catalog/pull/20

Reply:
@bzwei We can clear this TODO too
https://github.com/RedHatInsights/rhc-worker-catalog/blob/a795680f9b7a8e97dbf14e60279db1d1e06dcdee/internal/upload/upload.go#L33